### ISSUE TITLE
Can't submit to App Store using Xcode 15.3 with current minimum OS version

### ISSUE THREAD

**Issue by jonathansousa** [singular-labs/Singular-iOS-SDK#55]
In the 15.3 release of Xcode, Apple seems to have introduced a bug that surfaces during the App Store validation step with the error:

`Invalid Bundle. The bundle <App ID>/Frameworks/Singular.framework does not support the minimum OS Version specified in the Info.plist. (ID: 18a0de07-d9cc-4c96-afdf-2cb02327c59e)`

There is no way to fix this error in the app and is, apparently, a mistake in how minimum iOS versions in the app and packages are compared. The workarounds seem to be to downgrade Xcode to 15.2, or for the package maintainers to set the minimum iOS version to the impossible number of 100.

Firebase has encountered the same problem with their SDK and they have implemented the version 100 solution. https://github.com/firebase/firebase-ios-sdk/pull/12439#issuecomment-1994466302 They also have an Apple Feedback ID indicating their report of the problem.

While we await an Xcode fix from Apple, would it be possible to implement the version 100 fix so that the Singular SDK continues to work with the current Xcode version?

**Comment by remstos** [comment]
Hi, experiencing the same issue here 🙋‍♂️. Singular-iOS-SDK is the last framework to produce this warning in our project, so we can’t move our CI to Xcode 15.3 because of it.
Would be great to have the workaround mentioned above implemented 🙏

**Comment by adamkaump** [comment]
Same issue here. We're sticking with Xcode 15.2 for Xcode Cloud builds for now, but it would be great to get this addressed so we could get on the latest.

**Comment by kenanmamedoff** [comment]
![Screenshot 2024-03-24 at 13 23 07](https://github.com/singular-labs/Singular-iOS-SDK/assets/99484553/98cf5c60-c4c9-4865-a9c8-727bfa73b220)

Same here :)

**Comment by shefdanny** [comment]
Hey,
Thanks for bringing this up to our attention @jonathansousa .
We are looking into it and will update on any progress.

**Comment by pausanchezv** [comment]
Same issue here. We're sticking with Xcode 15.2 for Xcode Cloud builds for now. Xcode 15.3 has so many issues.

**Comment by jorgefrazaocosta** [comment]
Same here.

**Comment by jasdeepsaini** [comment]
Are there any updates on this issue? We have a large team that moved to Xcode 15.3 when it came back. Going back to Xcode 15.2 is not an option.

**Comment by jasdeepsaini** [comment]
I tried using 12.4.0, but I got compilation errors, which seemed related to not seeing the headers in the xcframework.  I decided to bring in the repo as a local SPM package and modified the Package.swift file. This works for me and should have the proper `PrivacyInfo.xcprivacy` files. I haven't tested it, but this should work in a fork too.

As a bonus, I was able to remove all of the `other link flags` and bridging headers specified in the Singular docs and the frameworks specified in this Repo's README.

I'm not sure if it's relevant, but I use the Singular SPM package in another SPM package.

Here's the package file I used with the v12.4.0 code.

```
// swift-tools-version:5.9

import PackageDescription

let package = Package(
    name: "Singular",
    platforms: [
        .iOS(.v15),
    ],
    products: [
        .library(
            name: "Singular",
            targets: ["Singular-Privacy", "Singular"]
        ),
    ],
    targets: [
        .target(
            name: "Singular-Privacy",
            path: "Singular",
            resources: [.copy("PrivacyInfo.xcprivacy")],
            linkerSettings: [
                .linkedLibrary("z"),
                .linkedLibrary("sqlite3.0"),
                .linkedFramework("SystemConfiguration"),
            ]
        ),
        .binaryTarget(
            name: "Singular",
            path: "Singular-SPM.xcframework"
        ),
    ]
)
```

**Comment by shefdanny** [comment]
fixed in version 12.3.3
